prow/apis/prowjobs/v1/types.go

```diff
@@ -286,10 +286,39 @@ type ReporterConfig struct {
 }
 
 type SlackReporterConfig struct {
-	Host              string         `json:"host,omitempty"`
-	Channel           string         `json:"channel,omitempty"`
-	JobStatesToReport []ProwJobState `json:"job_states_to_report,omitempty"`
-	ReportTemplate    string         `json:"report_template,omitempty"`
+	Host              *string         `json:"host,omitempty"`
```

Comment:
Why do these need to be string pointers? Is the empty string a valid config for any of  them? If not, just merging when its empty would make things easier IMHO

Reply:
changed them back to string